===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/unix/gsocket.cpp,v
retrieving revision 1.51
retrieving revision 1.51.2.1
diff -u -r1.51 -r1.51.2.1
--- wxWidgets/src/unix/gsocket.cpp	2005/09/19 15:18:27	1.51
+++ wxWidgets/src/unix/gsocket.cpp	2005/10/05 15:37:42	1.51.2.1
@@ -8,7 +8,7 @@
  *              Guillermo Rodriguez Garcia <guille@iies.es>
  * Purpose:     GSocket main Unix and OS/2 file
  * Licence:     The wxWindows licence
- * CVSID:       $Id: gsocket.cpp,v 1.51 2005/09/19 15:18:27 JS Exp $
+ * CVSID:       $Id: gsocket.cpp,v 1.51.2.1 2005/10/05 15:37:42 KH Exp $
  * -------------------------------------------------------------------------
  */
 
@@ -331,6 +331,9 @@
 
   assert(this);
 
+  /* Don't allow events to fire after socket has been closed */
+  gs_gui_functions->Disable_Events(this);
+
   /* If socket has been created, shutdown it */
   if (m_fd != INVALID_SOCKET)
   {
